site/src/components/mdx/mdx-page-content.tsx

```diff
@@ -139,16 +140,22 @@ export const MdxPageContent: FunctionComponent<MdxPageContentProps> = ({
               ? `#${headingAtScrollPosition.anchor}`
               : ""
           }`,
+          undefined,
+          {
+            scroll: false,
+          },
         );
       }
     };
 
-    window.addEventListener("scroll", onScroll);
+    setTimeout(() => {
+      window.addEventListener("scroll", onScroll);
+    }, 500);
```

Comment:
This is a simple fix for the stated bug. Next router's initial scroll to top functionality clashes with the scroll logic we have here. This adds a delay to wait for initial scrolling, that happens when a page is loaded, to be complete before running our scroll logic